### ISSUE TITLE
silent fail bug

### ISSUE THREAD

**Issue by ORESoftware** [algolia/algoliasearch-client-js#154]
hey there, small bug :) when calling this, your lib fails silently:

```javascript
   index.addObjects({}, function (err, content) {
                    if (err) {
                        // no error
                    }
                    else {
                        // so we are here
                    }
                });
```

as you can see we pass an object instead of an array

this lib fails silently, so I didn't know what the problem was. you can check for an array by using "obj instanceof Array"

but it shouldn't fail silently right? I fixed it by doing this, of course:

```javascript
   index.addObjects([{}], function (err, content) {
                    if (err) {
                        // no error
                    }
                    else {
                        // so we are here
                    }
                });
```

but anyway you should pass back an error in the callback if the dev makes the first mistake, thanks!

**Comment by vvo** [comment]
Hi, indeed we too often silently fail, I think I will do a pass soon on some obvious mistakes we should report first.

Thanks for reporting! If you have any other silent fail findings, report them again it's good.

**Comment by ORESoftware** [comment]
for sure np

**Comment by vvo** [comment]
Will release this today

**Comment by vvo** [comment]
Was released as 3.9.1 thanks for the idea
